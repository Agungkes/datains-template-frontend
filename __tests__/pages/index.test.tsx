import React from 'react';
import { render, screen } from '../testUtils';
import Home from '@pages/index';

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home />, {});
    expect(asFragment()).toMatchSnapshot();
  });

  it('ada tulisan welcome', () => {
    render(<Home />, {});
    // window.alert = jest.fn()
    // fireEvent.click(getByText('Test Button'))
    expect(screen.getByTestId('welcome-text')).toHaveTextContent(
      'Welcome to Next.js!',
    );
    // expect(window.alert).toHaveBeenCalledWith('With typescript and Jest')
  });
});
