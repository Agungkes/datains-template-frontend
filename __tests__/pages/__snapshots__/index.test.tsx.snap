// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <main>
      <h1
        data-testid="welcome-text"
      >
        Welcome to 
        <a
          href="https://nextjs.org"
        >
          Next.js!
        </a>
      </h1>
    </main>
  </div>
</DocumentFragment>
`;
